bm-persona/Common/DetailView/OpenTimesCardView.swift

```diff
@@ -0,0 +1,196 @@
+//
+//  OpenTimesCardView.swift
+//  bm-persona
+//
+//  Created by Shawn Huang on 7/4/20.
+//  Copyright © 2020 RJ Pimentel. All rights reserved.
+//
+
+import UIKit
+
+fileprivate let kViewMargin: CGFloat = 16
+
+// Displays the open times for the current day when collapsed and for all days when opened.
+class OpenTimesCardView: CollapsibleCardView {
+    var item: HasOpenTimes!
+
+    public init(item: HasOpenTimes, animationView: UIView, toggleAction: ((Bool) -> Void)? = nil, toggleCompletionAction: ((Bool) -> Void)? = nil) {
+        super.init()
+        self.item = item
+        super.setContents(collapsedView: collapsedView(), openedView: openedView(), animationView: animationView, toggleAction: toggleAction, toggleCompletionAction: toggleCompletionAction, leftIcon: clockIcon)
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
+    func collapsedView() -> UIView {
+        let defaultView = UIView()
+        defaultView.translatesAutoresizingMaskIntoConstraints = false
+        guard let weeklyHours = item.weeklyHours, weeklyHours.hoursForWeekday(DayOfWeek.weekday(Date())).count > 0 else {
+            return defaultView
+        }
+        var leftView: UIView?
+        var rightView: UIView?
+        
+        if let isOpen = item.isOpen {
+            if isOpen {
+                openTag.text = "Open"
+                openTag.backgroundColor = Color.openTag
+            } else {
+                openTag.text = "Closed"
+                openTag.backgroundColor = Color.closedTag
+            }
```

Comment:
There's also `TagView.open` and `TagView.closed` which has the text and colors you used here, so we could reduce this to: 
`leftView = isOpen ? TagView.open : TagView.closed`

(I think)